tests/test_sshfsmount.cpp

```diff
@@ -59,10 +67,52 @@ struct SshfsMount : public mp::test::SftpServerTest
         return request_exec;
     }
 
+    auto make_exec_to_check_commands(const command_vector& commands, std::string::size_type& remaining, command_vector::const_iterator& next_expected_cmd, std::string& output, bool& invoked)
```

Comment:
Is `invoked` actually used still?

Reply:
When a command is invoked, `invoke` makes the return channel to read. AFAICS, it's needed. Am I missing something?

Reply:
OK now I see where it is used… My only problem then is that invoked becomes `true` as soon as the first command is invoked, and it never goes false. So in theory an out-of-band read could be called, and the tests wouldn't catch that.

I think this is fine for now, but maybe we should add a TODO to make `test_command_execution` mimic this better, probably replacing `ssh_channel_read_timeout` on each command being invoked, and failing inside it if it's being called beyond the end of the returned data. Or at least resetting `invoked` to `false` when `remaining` goes 0.

Reply:
OK, now I see your point. I made a minor change, it should me much better now.